--- bc.cxx.origin	2025-04-01 20:40:06.370494832 +0900
+++ bc.cxx	2025-04-01 20:39:35.062399005 +0900
@@ -619,11 +619,11 @@
 #endif
                 break;
             case 3:
-                v[0] = 0;
+                v[0] = bc_vz1;
 #ifdef THREED
                 v[1] = 0;
 #endif
-                v[NDIMS-1] = bc_vz1;
+                v[NDIMS-1] = 0.0;
                 break;
             }
         }
